(DOCSP-32967): Move Edge Server content into new section (#581)
mongodb/docs-app-services: source/edge-server/configure.txt

   :depth: 2
   :class: singlecol

Edge Server is a "local" server that sits between your client devices and 
the MongoDB Atlas App. For most use cases, you'll provision hardware 
on-premises to host the Edge Server. While you're developing your Edge 
Server application, you may host it on a cloud server for convenience.

Client devices on-prem connect to Edge Server. Your Edge Server
connects to a MongoDB Atlas App. Clients do not connect to the MongoDB 

> Atlas App directly.

Reviewer: [suggestion]
add that simplified diagram here

Author: If we want to do this, we should get the image asset directly. I only have a screenshot I took from a Google doc. I'll see if I can track that down.

Author: Added it!

      
      .. tip:: Get started quickly with a template app

         You can use an Edge Server with any Flexible Sync-enabled 
         App Services App. For a quick proof-of-concept, you can 
         :ref:`create an app from a template <create-template-app>`. 
         Any of the starter templates whose name includes 
         :guilabel:`Atlas Device Sync Starter` can quick-start your 
         Edge Server proof-of-concept. If you create a template app from 
         the Atlas UI, choose :guilabel:`Real-time Sync`.
         
         These templates set up a Device Sync-enabled task tracker app with a 
         downloadable front-end client. The template app comes pre-configured 
         with permissions for the user to write their own Items to an ``Item`` 
         collection in a ``todo`` database. It also has 
         :ref:`Development Mode <development-mode>` and 
         :ref:`email/password authentication <email-password-authentication>` 
         enabled.

         On the :ref:`edge-server-connect` page, you can learn how to 
         modify the client to connect to the Edge Server.

   .. step:: Enable Device Sync

>       If you have not started with a template app, :ref:`enable Device Sync 

Reviewer: [entirely a suggestion]
```suggestion
      :ref:`Enable Device Sync <enable-sync>` with Flexible Sync.
      
      If you created a :ref:`template app <template-apps>`, 
      Device Sync with :ref:`Development Mode <development-mode>` is already enabled.
```

Author: 🤔  I'm undecided about this one because I know the option to create a Partition-Based Sync app is going away soon, and we want to just be talking about "Sync" with the understanding that Flexible Sync is the default. I'm not sure when that's going to happen, though. I'll ponder this.

      <enable-sync>` with Flexible Sync. 
      
   .. step:: Configure Data Access Permissions

>       Configure any :ref:`permissions <device-sync-compatibility>` your 

Reviewer: what's the deal with permissions already set for the template apps? i can't recall

Author: The template app already has permissions set: https://github.com/mongodb-university/realm-template-apps/blob/d46e24d8b9c88233c36944c414c970ba59aed2fa/sync-todo/v2/backend/data_sources/mongodb-atlas/default_rule.json

We used to have a thing in here to delete those permissions because Edge Server didn't support rules yet, but now it does so I removed the stuff about permissions relative to the template app. Are you thinking we should mention the template app's permissions here since we mention other things the template app pre-populates in the other steps?

      App needs to enforce data access rules.

   .. step:: Configure an Authentication Provider

      Edge Server supports these authentication providers:

      - :ref:`Anonymous Authentication <anonymous-authentication>`
      - :ref:`Email/Password Authentication <email-password-authentication>`
      - :ref:`Custom JWT Authentication <custom-jwt-authentication>`
      - :ref:`API Key Authentication <api-key-authentication>`

      Enable an authentication provider for clients to connect to the Edge Server.

>       .. note::

Reviewer: [entirely a suggestion]
```suggestion

```

Author: I think I want to set apart the stuff about the template app, because that's just a fast-track to get you going quickly. It's not inherently part of the instruction.

         
         If you created a :ref:`template app <template-apps>`, email/password 
         authentication is already enabled in the template app configuration.

.. _edge-server-configure-host:

> Configure the Host

Author: Renaming to Requirements.

-------------------------------------

Set up and run the Edge Server on the host.

.. include:: /includes/important-stop-tiered-sync-server.rst

.. procedure::

   .. step:: Get the Edge Server Code

      Use ``wget`` to get the current Edge Server code as a ``.tar`` file:

      .. code-block:: shell

         wget --content-disposition https://realm.mongodb.com/api/client/v2.0/tiered-sync/package/latest

      Unzip the tar to get the files.

      .. code-block:: shell

         tar -xvf *.tgz

      You now have an ``edge_server`` directory containing the server files.

   .. step:: Complete the Edge Server Configuration Details

      The ``edge_server`` directory contains a :file:`config.json` file you 
      must edit with the appropriate values to configure the server.

      .. code-block:: json

         {
           "clientAppId": "YOUR-APP-ID",
           "query": {
             "YOUR-COLLECTION-NAME": "YOUR-QUERY"
           },
           "cloudSyncServerAuthSecret": "YOUR-AUTH-SECRET",
           "hostname": "YOUR-EDGE-SERVER-DOMAIN"
         }

      .. list-table::
         :header-rows: 1

         * - Field name
           - Value

         * - ``clientAppId``

>            - The App ID of the App Services App you created above.

Reviewer: ```suggestion
           - The App ID of the App Services App you created in [link]previous step.
```
don't love the "above" here

         * - ``query``
           - An object containing one or more collections and queries that 
             define the collections and documents the Edge Server can 
             sync.

         * - ``YOUR-COLLECTION-NAME``
           - The name of the collection in your database where the Edge 
             Server should sync data. If you're using a 
             template app, this is the ``Item`` collection.
            
         * - ``YOUR-QUERY``

>            - This query determines what documents the Edge Server

Reviewer: ```suggestion
           - The query that determines what documents the Edge Server
```

             can sync with the collection. For example, ``truepredicate``
             means the Edge Server syncs all documents with the collection.
             You can use :ref:`Realm Query Language <rql>` to define this
             query, with the same caveats around :ref:`Flexible Sync RQL 
             Limitations <flexible-sync-rql-limitations>`.

>          * - ``cloudSyncServerAuthSecret``

Reviewer: should this have been mentioned already as a step? when should customers have done this? right after getting the App ID? as they're configuring the server?

>            - Give your :ref:`App Services App ID <find-app-id>` to your 

Reviewer: ```suggestion
           - The authorization secret that your Product or Account Representative generated for your App. 
```
only for consistency on these definitions. feels weird to have an instruction embedded in here, esp if it's the first time it's mentioned

>              Product or Account Representative. They will generate an

Reviewer: ```suggestion
             If you need an auth secret, give your :ref:`App Services App ID <find-app-id>` to your Product or Account Representative. They will generate an
```

Author: I've reworded this after adding a step to the prerequisites. Hopefully the new wording addresses this suggestion.

   :depth: 2
   :class: singlecol

Edge Server is a "local" server that sits between your client devices and 

> the MongoDB Atlas App. For most use cases, you'll provision hardware 

Reviewer 2: nit I would replace "Atlas App" with "Atlas App Services" here and in the next paragraph

.. _edge-server-hardware-requirements:

Hardware Requirements
~~~~~~~~~~~~~~~~~~~~~

Edge Server benchmarking is not yet complete, so minimum required hardware
is not yet available. 

.. _edge-server-install-and-configure:

Install and Configure the Edge Server
-------------------------------------

Set up and run the Edge Server on the host.

.. include:: /includes/important-stop-tiered-sync-server.rst

.. procedure::

   .. step:: Get the Edge Server Code

      Use ``wget`` to get the current Edge Server code as a ``.tar`` file:

      .. code-block:: shell

         wget --content-disposition https://realm.mongodb.com/api/client/v2.0/tiered-sync/package/latest

      Unzip the tar to get the files.

      .. code-block:: shell

         tar -xvf *.tgz

      You now have an ``edge_server`` directory containing the server files.

   .. step:: Complete the Edge Server Configuration Details

      The ``edge_server`` directory contains a :file:`config.json` file you 
      must edit with the appropriate values to configure the server.

      .. code-block:: json

         {
           "clientAppId": "YOUR-APP-ID",

>            "query": {

Reviewer 2: this is changing to default to `"query": "*"`

             "YOUR-COLLECTION-NAME": "YOUR-QUERY"
           },
           "cloudSyncServerAuthSecret": "YOUR-AUTH-SECRET",
           "hostname": "YOUR-EDGE-SERVER-DOMAIN",
           "tls" : {

>              "enabled": true,

Author: If this is "false" - what values are we expecting, if any, for the "certificates" field? Are we expecting the field to be present, but have no value? Are we expecting its value to be the object with "publicKeyPath" and "privateKeyPath" fields but no values or empty strings for those fields? Or are we expecting people to leave out the "certificates" key?

>              "enabled": true,

Reviewer 2: the default `config.json` will have enabled=false, along with placeholder certificates that point to nowhere. The user can either omit the `tls` bit entirely. In this case or if tls.enabled=false, certificates will be ignored entirely, regardless of whether it points to real certs

             "certificates": [
               {
                 "publicKeyPath": "YOUR_PUBLIC_KEY_PATH/cert.pem",
                 "privateKeyPath": "YOUR_PRIVATE_KEY_PATH/certkey.pem"
               }
             ]
           }
         }

      .. list-table::
         :header-rows: 1

         * - Field name
           - Value

         * - ``clientAppId``
           - The unique ID of your App Services App. For information about 
             finding the App ID, refer to :ref:`find-app-id`.

         * - ``query``

>            - An object containing one or more collections and queries that 

Reviewer 2: OR, it can be the literal string `"*"` to sync everything
